infra proxy list pages UI improvements

vinay033 opened the pull request:
### :nut_and_bolt: Description: What code changed, and why?

<!-- /!\ Please ensure that you are NOT disclosing any customer information without their consent /!\ -->
new UI for the list pages for read-only views (tabs UI, cookbooks, roles, environments, data bags and clients)
### :chains: Related Resources
https://github.com/chef/automate/issues/4663
### :+1: Definition of Done
I have added changes to the new UI list pages.
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

npm run serve:hab

To add data https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
Go to Infrastructure tab >> Chef Servers it's under Chef-Server feature flag.
```
### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
![clients](https://user-images.githubusercontent.com/12297653/106361489-326f1a00-6344-11eb-8a75-c8e88d32ca77.png)
![Databags](https://user-images.githubusercontent.com/12297653/106361491-3438dd80-6344-11eb-9e72-1168ad0b29da.png)
![Environments](https://user-images.githubusercontent.com/12297653/106361492-34d17400-6344-11eb-88da-d05757ca3e58.png)
![roles](https://user-images.githubusercontent.com/12297653/106361494-356a0a80-6344-11eb-86c9-8fa7de3d60d3.png)
![cookbooks](https://user-images.githubusercontent.com/12297653/106361495-3602a100-6344-11eb-9526-af2104da1326.png)

shaik80 commented:
can i know what  ui changes we are making, please describe in pr

vinay033 commented:
> can i know what ui changes we are making, please describe in pr

can you please check the related issue I have attached a screenshot.

vivekshankar1 commented:
hi @vinay033, i see the list menu and create databag mentioned in the issue. But there are few text changes as well in the pr. Is there any issue related or may i know what are we referring to for these text changes?

vinay033 commented:
> i see the list menu and create databag mentioned in the issue and those are removed. But there are few text changes as well in the pr. Is there any issue related or may i know what are we referring to for these text changes?

right now we are only targeting the read-only changes.. after the read-only release will add the create/edit/delete.

vivekshankar1 reviewed (approved):
Thank you. The changes look good and are as suggested in the issue screenshot.

shaik80 commented on components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html:
```diff
@@ -2,11 +2,14 @@
   <app-empty-state *ngIf="authFailure" moduleTitle="data bags" (resetKeyRedirection)="resetKeyTabRedirection($event)"></app-empty-state>
   <ng-container  *ngIf="!dataBagsListLoading && !authFailure">
+    <div class="head-container">
+      <!-- <p>Name of Data bags</p> -->
```
can you remove this comment

msorens commented:
@kalroy Why did you mark this PR blocked?

msorens commented on components/automate-ui/src/app/modules/infra-proxy/clients/clients.component.html:
```diff
@@ -4,8 +4,8 @@
-        <chef-tr>
-          <chef-th>Clients Name</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Clients</chef-th>
```
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
```

msorens commented on components/automate-ui/src/app/modules/infra-proxy/data-bags-list/data-bags-list.component.html:
```diff
@@ -2,11 +2,14 @@
+    <div class="head-container">
+      <!-- <p>Name of Data bags</p> -->
+      <chef-tr class="no_border_tr">
+        <chef-th class="no_border_th">Name of Data bags</chef-th>
```
```suggestion
        <chef-th class="no_border_th">Name</chef-th>
```

msorens commented on components/automate-ui/src/app/modules/infra-proxy/environments/environments.component.html:
```diff
@@ -4,8 +4,8 @@
-        <chef-tr>
-          <chef-th>Environments Name</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Environments</chef-th>
```
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
```

msorens commented on components/automate-ui/src/app/modules/infra-proxy/cookbooks/cookbooks.component.html:
```diff
@@ -4,9 +4,9 @@
-          <chef-th>Cookbook Name</chef-th>
-          <chef-th>Cookbook Version</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Cookbook</chef-th>
```
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
```

msorens commented on components/automate-ui/src/app/modules/infra-proxy/infra-roles/infra-roles.component.html:
```diff
@@ -4,10 +4,10 @@
-          <chef-th>Description</chef-th>
-          <chef-th>Environments</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Role</chef-th>
+          <chef-th class="no_border_th">Description of Roles</chef-th>
+          <chef-th class="no_border_th">Environments of Roles</chef-th>
```
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
          <chef-th class="no_border_th">Description</chef-th>
          <chef-th class="no_border_th">Environment</chef-th>
```

msorens commented on components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html:
```diff
@@ -4,9 +4,9 @@
-          <chef-th>Policy Name</chef-th>
-          <chef-th>Policy Group</chef-th>
+        <chef-tr class="no_border_tr">
+          <chef-th class="no_border_th">Name of Policies</chef-th>
+          <chef-th class="no_border_th">Group of Policies </chef-th>
```
```suggestion
          <chef-th class="no_border_th">Name</chef-th>
          <chef-th class="no_border_th">Group</chef-th>
```

msorens commented on components/automate-ui/src/app/modules/infra-proxy/policy-files/policy-files.component.html:
```diff
@@ -4,9 +4,9 @@
     <chef-table>
       <chef-thead>
-        <chef-tr>
-          <chef-th>Policy Name</chef-th>
```
Neither your PR preamble nor the associated issue mentions this policy page--so is this really supposed to be included?

vinay033 replied:
yes for now.

msorens reviewed (changes requested):
There is apparently no UX specification for this issue and PR, so I will make my recommendation: let's omit the "of x" for every header.
It does not add anything to use headers Name of Role, Description of Role, Environment of Role; it only makes it cluttered.
See my specific comments below.
Note that I have also made all such terms singular; previously it was inconsistent with some singular and some plural.

vinay033 commented:
> There is apparently no UX specification for this issue and PR, so I will make my recommendation: let's omit the "of x" for every header.
> It does not add anything to use headers Name of Role, Description of Role, Environment of Role; it only makes it cluttered.
> See my specific comments below.
> Note that I have also made all such terms singular; previously it was inconsistent with some singular and some plural.
I did changes as per the UX suggestions but now I have changed it to `Name` only and updated the PR.

kalroy commented:
> @kalroy Why did you mark this PR blocked?

I thought it was blocked by an API PR waiting to be merged. Since it is not the case, I am moving it out to Review.

chaitali-mane commented:
While testing found minor CSS issues like more column width in list data, More space for list heading.

msorens commented:
@chaitali-mane Please create a new GitHub issue for the CSS issues you indicated.

chaitali-mane commented:
> @chaitali-mane Please create a new GitHub issue for the CSS issues you indicated.

Created issue for the UI chages #4750 and raised PR with changes #4754 against issue.
